PMSCore/EntityJsonConverter.cs

```diff
@@ -0,0 +1,117 @@
+using System.Data;
+using System.Net.NetworkInformation;
+using System.Reflection.Metadata;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace PMSCore;
+
+public class EntityJsonConverter : JsonConverter<Entity>
+{
+        public override Entity? Read(
+            ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType != JsonTokenType.StartObject)
+            {
+                throw new JsonException();
+            }
+            
+            reader.Read();
+            if (reader.TokenType != JsonTokenType.PropertyName)
+            {
+                throw new JsonException();
+            }
+            
+            string? propertyName = reader.GetString();
+            if (propertyName != "Type")
+            {
+                throw new JsonException();
+            }
+            
+            reader.Read();
+            
+            var entityType = reader.GetString();
+            Entity entity = entityType switch
+            {
+                "Project" => new  Project("default", null, default, default, default),
+                "Task" => new Task("default", null, default, default, default),
+                _ => throw new JsonException()
+            };
+            entity.Type = Enum.Parse<EntityType>(entityType);
+            while (reader.Read())
+            {
+                if (reader.TokenType == JsonTokenType.EndObject)
+                {
+                    return entity;
+                }
+            
+                if (reader.TokenType == JsonTokenType.PropertyName)
+                {
+                    propertyName = reader.GetString();
+                    reader.Read();
+                    var value = reader.GetString();
+                    switch (propertyName)
+                    {
+                        case "Id":
+                            if (value == null)
+                            {
+                                return null;
+                            }
+                            entity.Id = value;
+                            break;
+                        case "Description":
+                            entity.Description = value;
+                            break;
+                        case "Status":
+                            entity.Status = Enum.Parse<EntityStatus>(value!); 
+                            break;
+                        case "Priority":
+                            entity.Priority =  Enum.Parse<EntityPriority>(value!); 
+                            break;
+                        case "CreatedDate":
+                            entity.CreatedDate = DateTime.Parse(value!);
```

Comment:
entity.CreatedDate = DateTime.Parse(value!); 

might be better as 

entity.CreatedDate = string.IsNullOrEmpty(value) ? DateTime.MinValue : DateTime.Parse(value);

what do you think?